File: Third_Year/DS/Practicas/P1/Filtros/CadenaFiltro.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

#include "CadenaFiltro.h"

void CadenaFiltro::addFiltro(Filtro * f){
    filtros.push_back(f);
}

void CadenaFiltro::ejecutar(Alquiler & alq){
    for(int i = 0; i < filtros.size(); i++){
        filtros[i]->ejecutar(alq);
    }
}

File: Third_Year/DS/Practicas/P4/codigo_flutter/CadenaFiltro.dart
import 'Filtro.dart';
import 'Alquiler.dart';

class CadenaFiltro{
  List<Filtro> _filtros = [];

  void addFiltro(Filtro f){

    _filtros.add(f);
  }

  void ejecutar(Alquiler alq){
    for(var i = 0; i < _filtros.length ;i++){
      _filtros[i].ejecutar(alq);
    }
  }
}